pkgs/applications/misc/coolreader/default.nix

```diff
@@ -2,25 +2,34 @@
   stdenv,
   mkDerivation,
   fetchFromGitHub,
+  fetchpatch,
   cmake,
   pkg-config,
   lib,
   qttools,
   fribidi,
   libunibreak,
+  zstd,
 }:
 
 mkDerivation rec {
   pname = "coolreader";
-  version = "3.2.57";
+  version = "3.2.58";
 
   src = fetchFromGitHub {
     owner = "buggins";
     repo = "coolreader";
     rev = "cr${version}";
-    sha256 = "sha256-ZfgaLCLvBU6xP7nx7YJTsJSpvpdQgLpSMWH+BsG8E1g=";
+    sha256 = "sha256-DUcYUFxPPSPvoEUEbKYEAGxFeFGQCfOFA0+SegoC4oI=";
   };
 
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/buggins/coolreader/commit/fc4baf7cbb71ba7a47b174870dacf0e4ec93b6bf.patch";
+      hash = "sha256-CehPJ85GZXhibUZlvrYdR2M1DuZ1LTrMwvRmvaFsRb4=";
+    })
+  ];
+
```

Comment:
Can you vendor the patch and add a comment to the pr (https://github.com/buggins/coolreader/pull/365)?

Reply:
yes